control/tests/statesp_test.py

```diff
@@ -519,6 +563,177 @@ def test_lft(self):
+        D = np.array([[13.0 +1.0j, -12.0 +1.0j],
+                    [3.0 +2.0j, -5.0 +11.0j]])
+
+        sys = self.sysC322 + self.sysC222
+
+        np.testing.assert_array_almost_equal(sys.A, A)
+        np.testing.assert_array_almost_equal(sys.B, B)
+        np.testing.assert_array_almost_equal(sys.C, C)
+        np.testing.assert_array_almost_equal(sys.D, D)
+
+    def test_subtract_ss_complex(self):
+        """Subtract two complex MIMO systems."""
+
+        A = np.array([[6.0 +9.0j, 8.0 +9.0j, -4.0 -7.0j, 0.0, 0.0],
+                    [8.0 -7.0j, 3.0, 1.0 -1.0j, 0.0, 0.0],
+                    [-7.0 +9.0j, -8.0 +6.0j, 9.0 +8.0j, 0.0, 0.0],
+                    [0.0, 0.0, 0.0, -4.0 -7.0j, 3.0 +9.0j],
+                    [0.0, 0.0, 0.0, 3.0, -6.0 -3.0j]])
+        B = np.array([[6.0 +3.0j, -9.0 -2.0j],
+                    [9.0 +5.0j, 7.0 +3.0j],
+                    [3.0 +5.0j, 8.0 -6.0j],
+                    [2.0, 5.0 +7.0j],
+                    [-5.0 +4.0j, -5.0 +9.0j]])
+        C = np.array([[4.0 +4.0j, -4.0 +9.0j, -8.0 -1.0j, -1.0 -6.0j, 7.0 -6.0j],
+                    [-9.0 -3.0j, -9.0 -9.0j, 6.0 -2.0j, 7.0 +5.0j, 5.0 +5.0j]])
+        D = np.array([[-3.0 -3.0j, 7.0j],
+                    [9.0 +4.0j, 5.0 -1.0j]])
+
+        sys = self.sysC322 - self.sysC222
+
+        np.testing.assert_array_almost_equal(sys.A, A)
+        np.testing.assert_array_almost_equal(sys.B, B)
+        np.testing.assert_array_almost_equal(sys.C, C)
+        np.testing.assert_array_almost_equal(sys.D, D)
+
+    def test_multiply_ss_complex(self):
+        """Multiply two complex MIMO systems."""
+
+        A = np.array([[6.0 +9.0j, 8.0 +9.0j, -4.0 -7.0j, 41.0 +98.0j, -25.0 +70.0j],
+                    [8.0 -7.0j, 3.0, 1.0 -1.0j, -55.0 +3.0j, -113.0 -31.0j],
+                    [-7.0 +9.0j, -8.0 +6.0j, 9.0 +8.0j, -113.0 +25.0j, -121.0 -27.0j],
+                    [0.0, 0.0, 0.0, -4.0 -7.0j, 3.0 +9.0j],
+                    [0.0, 0.0, 0.0, 3.0, -6.0 -3.0j]])
+        B = np.array([[67.0 +51.0j, 30.0 -80.0j],
+                    [44.0 +42.0j, -92.0 -30.0j],
+                    [-16.0 +56.0j, -7.0 +39.0j],
+                    [2.0, 5.0 +7.0j],
+                    [-5.0 +4.0j, -5.0 +9.0j]])
+        C = np.array([[4.0 +4.0j, -4.0 +9.0j, -8.0 -1.0j, 73.0 +31.0j, 21.0 +47.0j],
+                    [-9.0 -3.0j, -9.0 -9.0j, 6.0 -2.0j, 13.0 +4.0j, -35.0 -10.0j]])
+        D = np.array([[64.0 -4.0j, -27.0 -65.0j],
+                    [47.0 +21.0j, -57.0 -61.0j]])
+
+        sys = self.sysC322 * self.sysC222
+
+        np.testing.assert_array_almost_equal(sys.A, A)
+        np.testing.assert_array_almost_equal(sys.B, B)
+        np.testing.assert_array_almost_equal(sys.C, C)
+        np.testing.assert_array_almost_equal(sys.D, D)
+
+    def test_evalfr_complex(self):
+        """Evaluate the frequency response at one frequency."""
+
+        resp = [[4.6799374736968655 -34.9854626345217383j,
```

Comment:
Please be easier on the precision of your reference values. This will almost certainly fail on some build systems due to rounding errors or depending on the used BLAS/LAPACK implementation.